src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStaticMethods.cs

```diff
@@ -61,11 +61,9 @@ internal static int GetCurrentProcessIdForTdsLoginOnly()
         {
             if (s_currentProcessId == NoProcessId)
             {
-                // In ProjectK\CoreCLR we don't want to take a dependency on an assembly
-                // just to grab the real Process Id that the server doesn't really use
-                // So, instead, pick a random number and use that for all connections
-                Random rand = new Random();
-                int processId = rand.Next();
+                // Pick up the process Id from the current process instead of randomly generating it.
+                // This would be helpful while tracing application related issues.
+                int processId = System.Diagnostics.Process.GetCurrentProcess().Id;
                 Threading.Interlocked.CompareExchange(ref s_currentProcessId, processId, NoProcessId);
```

Comment:
This CompareExchange isn't really necessary: it could just be:
```C#
Volatile.Write(ref s_currentProcessId, processId);
```
as if there are multiple threads that race to initialize this, they're all going to be initializing it to the same value, since the current process' ID will not change.

Reply:
Agreed, I have updated code to use `Volatile.Write` instead.